watch.go

```diff
@@ -21,7 +23,23 @@ type Event struct {
 	// Path to the data that changed
 	Path string
 	// Data that changed
-	Data interface{}
+	Data    interface{}
+	payload string
+}
+
+func (e Event) Value(v interface{}) (path string, err error) {
```

Comment:
If you decide to deprecate the current solution (with the
map[string]interface{} style), then you can save CPU time and RAM usage.
But then, the path will not be extracted and will only be available through
the "Value" method.

█  Jacques

On 31 July 2016 at 21:00, Steven Berlanga <notifications@github.com> wrote:

> In watch.go
> <https://github.com/zabawaba99/firego/pull/50#discussion_r72908982>:
>
> > @@ -21,7 +23,23 @@ type Event struct {
> >  	// Path to the data that changed
> >  	Path string
> >  	// Data that changed
> > -	Data interface{}
> > +	Data    interface{}
> > +	payload string
> > +}
> > +
> > +func (e Event) Value(v interface{}) (path string, err error) {
>
> Why would it matter to return the path here? It's already available
> through the Event struct.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zabawaba99/firego/pull/50/files/91637d68c23bcd6761930b23882b18621b759365#r72908982>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAB2iEpgNCe7MkwONavjYuGrq89V_e6Qks5qbPDAgaJpZM4JYHTS>
> .
>